query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -0,0 +1,634 @@
+                if (requestTimeMs > slowQueryThreshold) {
+                  try {
+                    LOG.warn(
+                        "Query Execution time: {} ms, sqlQuery: {}, queryRequest: {}, executionStats: {}",
+                        requestTimeMs,
+                        pql.getKey(),
+                        protoJsonPrinter.print(request),
+                        "Stats not available");
+                  } catch (InvalidProtocolBufferException ignore) {
+                  }
+                }
+              });
+    } catch (Throwable error) {
+      return Observable.error(error);
+    }
+  }
+
+  @Nonnull
+  private QueryRequest rewriteRequestWithViewFiltersApplied(
+      QueryRequest request, Map<String, TableColumnFilter> columnFilterMap) {
+
+    Filter newFilter = removeViewColumnFilter(request.getFilter(), columnFilterMap);
+    QueryRequest.Builder builder = QueryRequest.newBuilder(request).clearFilter();
+    if (!Filter.getDefaultInstance().equals(newFilter)) {
+      builder.setFilter(newFilter);
+    }
+    return builder.build();
+  }
+
+  private Filter removeViewColumnFilter(
+      Filter filter, Map<String, TableColumnFilter> columnFilterMap) {
+    if (filter.getChildFilterCount() > 0) {
+      // Recursively try to remove the filter and eliminate the null nodes.
+      Set<Filter> newFilters =
+          filter.getChildFilterList().stream()
+              .map(f -> removeViewColumnFilter(f, columnFilterMap))
+              .filter(f -> !Filter.getDefaultInstance().equals(f))
+              .collect(Collectors.toCollection(LinkedHashSet::new));
+
+      if (newFilters.isEmpty()) {
+        return Filter.getDefaultInstance();
+      } else if (newFilters.size() == 1) {
+        return Iterables.getOnlyElement(newFilters);
+      } else {
+        return Filter.newBuilder(filter).clearChildFilter().addAllChildFilter(newFilters).build();
+      }
+    } else {
+      return rewriteLeafFilter(filter, columnFilterMap);
+    }
+  }
+
+  private Filter rewriteLeafFilter(
+      Filter queryFilter, Map<String, TableColumnFilter> columnFilterMap) {
+    TableColumnFilter tableColumnFilter =
+        columnFilterMap.get(
+            getLogicalColumnName(queryFilter.getLhs()).orElseThrow(IllegalArgumentException::new));
+    // If the RHS of both the view filter and query filter match, return empty filter.
+    if (tableColumnFilter != null
+        && isEquals(tableColumnFilter.getValues(), queryFilter.getRhs())) {
+      return Filter.getDefaultInstance();
+    }
+
+    // In every other case, retain the query filter.
+    return queryFilter;
+  }
+
+  Observable<Row> convert(ResultSet resultSet, LinkedHashSet<String> selectedAttributes)
+      throws SQLException {
+    List<Builder> rowBuilderList = new ArrayList<>();
+    if (resultSet.next()) {
+      // Postgres has different Response format for selection and aggregation/group by query.
```

Comment:
I think this is the case with pinot. Postgres will always return as a list of columns in response, right?

Reply:
I think we need a single parse function in Postgres.

Reply:
That part is correct. Will fix it in follow up PR.